test/test_redis_asyncio.py

```diff
@@ -0,0 +1,39 @@
+import pytest
+import redis
+import redis.client
+from packaging.version import Version
+
+REDIS_VERSION = Version(redis.__version__)
+
+below_redis_4_2 = pytest.mark.skipif(
+    REDIS_VERSION >= Version("4.2.0"),
+    reason="Test is only applicable to redis-py below 4.2.0",
+)
+redis4_2_and_above = pytest.mark.skipif(
+    REDIS_VERSION < Version("4.2.0"),
+    reason="Test is only applicable to redis-py 4.2.0 and above",
+)
```

Comment:
I thought about tackling #17 as well for this to reduce duplication, but thought it would be best to limit the scope.

Thinking about it now, this whole test function should actually be fine in `test_fakeredis.py`, I thought I had a reason to split it out, but I can't remember it on the top of my mind right now.